.github/workflows/docs.yml

```diff
@@ -0,0 +1,43 @@
+name: Rustdoc
+on:
+  push:
+    branches: [main]
+  pull_request:
```

Comment:
Not having the `main` branch here means that if some doc fails, we will know once it's merged in `main`? Reformulating the question, do we want to check the doc integrity for each PR before merging it?

Reply:
Good point. We should know it would fail before merging for sure!

Reply:
This CI actually builds the docs on PRs against `main`, it just doesn't publish: https://github.com/foss3/runtime-pallet-library/actions/runs/6550910806/job/17790998211?pr=4

Reply:
Right! thanks